drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/SettingsPresenter.java

```diff
@@ -92,23 +94,38 @@ public void setSaveCommand(Command saveCommand) {
         this.saveCommand = saveCommand;
     }
 
+    @Override
+    public void setSaveEnabled(boolean toSet) {
+        saveEnabled = toSet;
+        if (!saveEnabled) {
+            view.removeSaveButton();
+        } else {
+            view.restoreSaveButton();
+        }
+    }
+
     @Override
     public void onSaveButton(String scenarioType) {
-        simulationDescriptor.setSkipFromBuild(view.getSkipFromBuild().isChecked());
-        switch (ScenarioSimulationModel.Type.valueOf(scenarioType)) {
-            case RULE:
-                saveRuleSettings();
-                break;
-            case DMN:
-                saveDMNSettings();
-                break;
+        if (saveCommand != null) {
+            simulationDescriptor.setSkipFromBuild(view.getSkipFromBuild().isChecked());
+            switch (ScenarioSimulationModel.Type.valueOf(scenarioType)) {
+                case RULE:
+                    saveRuleSettings();
+                    break;
+                case DMN:
+                    saveDMNSettings();
+                    break;
```

Comment:
Just  a small suggestion to add a default with new IlligalStateException in case if we would add some new scenario type.

Reply:
Hi @dupliaka 
in other contexts I would do exactly as per your suggestion.
Here, throwing an exception could lead to an unexpected (or difficult to manage) ui behavior, so for the moment being I'm adding an empty default.
What should be actually tested (but it is not doable in this current PR) is that this method is _never_ invoked with a not-existing type of scenario.

Reply:
Thanks for an explanation, I will keep it in mind